src/app/features/worklog/worklog-export/worklog-export.util.ts

```diff
@@ -4,122 +4,20 @@ import { msToString } from '../../../ui/duration/ms-to-string.pipe';
-    if ((groupBy === WorklogGrouping.PARENT && task.parentId !== null)
-      || (groupBy === WorklogGrouping.TASK && task.subTaskIds.length > 0)
-    ) {
-      return taskGroups;
-    }
-
-    switch (groupBy) {
-      case WorklogGrouping.DATE:
-        if (!task.timeSpentOnDay) {
-          return {};
-        }
-        const numDays = Object.keys(task.timeSpentOnDay).length;
-        Object.keys(task.timeSpentOnDay).forEach(day => {
-          taskGroups[day] = {
-            dates: [day],
-            tasks: [task],
-            notes: (task.notes) ? [task.notes] : [],
-            projects: (task.projectId) ? [task.projectId] : [],
-            tags: task.tagIds,
-            workStart: startTimes[day],
-            workEnd: endTimes[day],
-            timeSpent: 0,
-            timeEstimate: 0
-          };
-          if (!task.subTaskIds || task.subTaskIds.length === 0) {
-            taskGroups[day].timeSpent = task.timeSpentOnDay[day];
-            taskGroups[day].timeEstimate = task.timeEstimate / numDays;
-          }
-        });
-        break;
-      case WorklogGrouping.PARENT:
-      case WorklogGrouping.TASK:
-        taskGroups[task.id] = createEmptyGroup();
-
-        // by design subtasks don't have tags, so we must set its parent's tags
-        if (task.parentId !== null) {
-          taskGroups[task.id].tags = (tasks.find(t => t.id === task.parentId) as WorklogTask).tagIds;
-        } else {
-          taskGroups[task.id].tags = task.tagIds;
-        }
-
-        taskGroups[task.id].tasks = [task];
-        taskGroups[task.id].notes = (task.notes) ? [task.notes] : [];
-        taskGroups[task.id].projects = (task.projectId) ? [task.projectId] : [];
-        taskGroups[task.id].dates = Object.keys(task.timeSpentOnDay);
-        taskGroups[task.id].timeEstimate = task.timeEstimate;
-        taskGroups[task.id].timeSpent = Object.values(task.timeSpentOnDay).reduce((acc, curr) => acc + curr, 0);
-        break;
-      default: // group by work log (don't group at all)
-        Object.keys(task.timeSpentOnDay).forEach(day => {
-          const groupKey = task.id + '_' + day;
-          taskGroups[groupKey] = createEmptyGroup();
-          taskGroups[groupKey].tasks = [task];
-          taskGroups[groupKey].notes = (task.notes) ? [task.notes] : [];
-          taskGroups[groupKey].projects = (task.projectId) ? [task.projectId] : [];
-          taskGroups[groupKey].tags = task.tagIds;
-          taskGroups[groupKey].dates = [day];
-          taskGroups[groupKey].workStart = startTimes[day];
-          taskGroups[groupKey].workEnd = endTimes[day];
-          taskGroups[groupKey].timeEstimate = task.subTaskIds.length > 0 ? 0 : task.timeEstimate;
-          taskGroups[groupKey].timeSpent = task.subTaskIds.length > 0 ? 0 : task.timeSpentOnDay[day];
-        });
-    }
-    return taskGroups;
-  };
-
+export const createRows = (data: WorklogExportData, groupBy: WorklogGrouping): RowItem[] => {
```

Comment:
I would love to see a short introductory comment on what this does and how, because I am pretty sure I will quickly forget this :)